tests/pass/permissive_provenance_sb.rs

```diff
@@ -0,0 +1,62 @@
+// compile-flags: -Zmiri-permissive-provenance
+
+
+// random agglomeration test
+use core::cell::UnsafeCell;
+
+fn not_so_innocent(x: &mut UnsafeCell<u32>) {
+    let x_raw: *mut UnsafeCell<u32> = x as *mut UnsafeCell<u32>;
+    x_raw as usize;
+}
+
+fn main() {
```

Comment:
Please add this as a new testcase to the existing file tests/pass/stacked-borrows/int-to-ptr.rs (and add `-Zmiri-permissive-provenance` to that file).